bdsx/bds/implements.ts

```diff
@@ -839,8 +839,23 @@ BlockActor.prototype.load = function(tag) {
     }
 };
 BlockActor.prototype.setChanged = procHacker.js("BlockActor::setChanged", void_t, {this:BlockActor});
+BlockActor.prototype.setCustomName = procHacker.js("BlockActor::setCustomName", void_t, {this:BlockActor}, CxxString);
 BlockActor.prototype.getContainer = makefunc.js([0x380], Container, { this: BlockActor });
 BlockActor.prototype.getType = procHacker.js("BlockActor::getType", int32_t.ref(), {this:BlockActor});
+BlockActor.prototype.getPosition = procHacker.js("BlockActor::getPosition", BlockPos, {this:BlockActor});
+BlockActor.prototype.updateClientSide = function(player: ServerPlayer): void {
+    const pk = BlockActorDataPacket.allocate();
+    const nbtData = this.allocateAndSave();
+    pk.pos.set(this.getPosition());
+    pk.data.construct(nbtData);
+    player.sendNetworkPacket(pk);
+    nbtData.destruct();
```

Comment:
the return value of `allcoateAndSave()` needs to call `dispose()` not `destruct()`.
`destruct()` does not `free` the instance's own memory.
but `allocate` things use `malloc` for allocating own memory.

and don't use `construct` for the already constructed field or use `destruct` before using it.